File: modules/common/src/main/java/com/sygic/modules/common/di/ContextModule.java
package com.sygic.modules.common.di;

import android.app.Application;
import android.content.Context;
import androidx.fragment.app.Fragment;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;


@Module
public class ContextModule {

    private final Application application;

    public ContextModule(final Fragment fragment) {
        application = fragment.requireActivity().getApplication();
    }

    @Singleton
    @Provides
    Context provideContext() {
        return application;
    }
}

File: modules/common/src/main/java/com/sygic/modules/common/di/ContextModule.kt
package com.sygic.modules.common.di

import android.app.Application
import android.content.Context
import androidx.fragment.app.Fragment
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ContextModule(fragment: Fragment) {

    private val application: Application = fragment.requireActivity().application

    @Singleton
    @Provides
    fun provideContext(): Context = application
}